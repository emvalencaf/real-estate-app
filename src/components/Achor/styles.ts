import styled, { css } from "styled-components";

export const Achor = styled.a`
	${({ theme }) => css`
		display: flex;
		flex-direction: row;
		align-items: center;
		justify-content: center;
		gap: ${theme.spacings.tiny};
		color: ${theme.colors.secondary};
		font-size: ${theme.fonts.sizes.normal};
		text-decoration: none;
		transition: all ${theme.transitions.normal} ease-in-out;

		> svg {
			height: ${theme.fonts.sizes.medium};
			width: ${theme.fonts.sizes.medium};
			transition: all ${theme.transitions.slow} ease-in-out;
		}

		&:hover {
			color: ${theme.colors.tertiary};
			filter: brightness(220%);

			> svg {
				transform: scale(1.3);
			}
		}
	`}
`;
