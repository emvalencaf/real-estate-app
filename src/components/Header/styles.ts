import styled, { css } from "styled-components";

export const Header = styled.header`
	${({ theme }) => css`
		display: flex;
		width: 100%;
		justify-content: space-between;
		align-items: center;
		padding: 0px ${theme.spacings.xsmall};
		margin: auto;
		background-color: ${theme.colors.deepWhite};
		border-bottom: 1px solid transparent;
		box-shadow: 0px 0px 10px black;
	`}
`;
