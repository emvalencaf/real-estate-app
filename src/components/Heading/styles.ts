import styled, { css, DefaultTheme } from "styled-components";
import { HeadingProps } from ".";

const titleSize = {
	small: (theme: DefaultTheme, size: string) => css`
		font-size: ${theme.fonts.sizes.medium};
		${mediaFont(theme, size)};
	`,
	medium: (theme: DefaultTheme, size: string) => css`
		font-size: ${theme.fonts.sizes.large};
		${mediaFont(theme, size)};
	`,
	big: (theme: DefaultTheme, size: string) => css`
		font-size: ${theme.fonts.sizes.xlarge};
		${mediaFont(theme, size)};
	`,
	huge: (theme: DefaultTheme, size: string) => css`
		font-size: ${theme.fonts.sizes.xhuge};
		${mediaFont(theme, size)};
	`,
};

const mediaFont = (theme: DefaultTheme, size: string) => {
	const fontSize = {
		lteSmall: {
			huge: theme.fonts.sizes.large,
			big: theme.fonts.sizes.medium,
			medium: theme.fonts.sizes.small,
			small: theme.fonts.sizes.xsmall,
		},
		lteMedium: {
			huge: theme.fonts.sizes.xlarge,
			big: theme.fonts.sizes.large,
			medium: theme.fonts.sizes.medium,
			small: theme.fonts.sizes.small,
		},
	};
	return css`
		@media ${theme.media.lteOrEqMedium} {
			font-size: ${fontSize.lteSmall[size]};
		}
		@media ${theme.media.lteOrEqMedium} {
			font-size: ${fontSize.lteMedium[size]};
		}
	`;
};

const titleCase = (uppercase: boolean) => css`
	text-transform: ${uppercase ? "uppercase" : "none"};
`;

const titleWeight = (weight: string) => css`
	font-weight: ${weight};
`;

export const Title = styled.h1<HeadingProps>`
	${({ theme, size, uppercase, color, weight }) => css`
		color: ${theme.colors[color]};
		line-height: 1.2;
		${titleSize[size](theme, size)};
		${titleCase(uppercase)};
		${titleWeight(weight)}
	`}
`;
