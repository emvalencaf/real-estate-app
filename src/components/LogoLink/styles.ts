import styled, { css } from "styled-components";

export const Wrapper = styled.div`
	${({ theme }) => css`
		& picture,
		img {
			height: ${theme.fonts.sizes.xsmall};
		}
	`}
`;
