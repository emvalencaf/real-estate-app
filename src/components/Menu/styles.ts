import styled, { css } from "styled-components";

export const Wrapper = styled.div`
	${({ theme }) => css``}
`;

export const Nav = styled.nav`
	${({ theme }) => css``}
`;

export const Ul = styled.ul`
	${({ theme }) => css`
		display: flex;
		justify-content: space-between;
		align-items: center;
	`}
`;
