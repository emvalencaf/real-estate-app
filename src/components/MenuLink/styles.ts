import styled, { css } from "styled-components";

export const ListMenu = styled.li`
	${({ theme }) => css`
		list-style: none;
	`}
`;
