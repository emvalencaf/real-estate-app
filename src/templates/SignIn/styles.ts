import styled, { css } from "styled-components";

export const Wrapper = styled.main`
	${({ theme }) => css`
		height: 100vh;
		background-color: rgba(240, 253, 244);
	`}
`;

export const Section = styled.section`
	${({ theme }) => css`
		@media ${theme.media.gteOrEqLarge} {
			& picture > img,
			& picture {
				width: 50%;
			}
			& picture {
				border-right: ${theme.spacings.tiny} solid;
			}
		}
		@media ${theme.media.lteOrEqMedium} {
			& picture > img,
			& picture {
				width: 67%;
			}
			& picture {
				border-right: ${theme.spacings.tiny} solid;
			}
		}
		@media ${theme.media.lteOrEqSmall} {
			& picture > img {
				width: 100%;
			}
		}

		& picture > img {
			border-radius: ${theme.spacings.small};
		}
	`}
`;
