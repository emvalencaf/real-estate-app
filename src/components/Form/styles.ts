import styled, { css, DefaultTheme } from "styled-components";
import { Button } from "../Button/styles";

type AlertProps = {
	theme: DefaultTheme;
	isSuccess?: boolean;
	visible?: boolean;
};

export const Form = styled.form`
	${({ theme }) => css`
		display: flex;
		flex-direction: column;
		justify-content: center;
		gap: ${theme.spacings.medium};
		width: 100%;
		max-width: 100rem;
		padding: ${theme.spacings.small};
		border-radius: 0rem 2rem 2rem;
		background-color: ${theme.colors.deepWhite};
		}
	`}
`;

export const ContainerButton = styled.div`
	${() => css`
		align-self: flex-end;
	`}
`;

export const Alert = styled.div<AlertProps>`
	${({ theme, isSuccess = false, visible = false }) => css`
		position: fixed;
		display: ${visible ? "flex" : "none"};
		top: 0;
		left: 0;
		right: 0;
		align-items: center;
		justify-content: center;
		color: ${theme.colors.quaternary};
		background-color: ${isSuccess
			? theme.colors.success
			: theme.colors.warning};
		padding: ${theme.spacings.small};
		border-radius: 0 0 0.4rem 0.4rem;
		z-index: ${theme.layers.layer7};
		transition: all ${theme.transitions.slow} ease-in-out;
		gap: ${theme.spacings.tiny};

		& a {
			color: ${theme.colors.quaternary};
		}

		> span {
			display: flex;
			gap: ${theme.spacings.xxsmall};
		}

		& ${Button} {
			position: fixed;
			align-self: flex-end;
			padding: 0px;
			margin: 0px;
			top: 0.2rem;
			right: 0.2rem;
			> svg {
				width: 2rem;
				height: 2rem;
			}
		}
	`}
`;

export const CloseButton = styled.span`
	${({ theme }) => css`
		position: fixed;
		align-self: flex-end;
		padding: 0px;
		margin: 0px;
		top: 0.2rem;
		right: 0.2rem;
		background-color: ${theme.colors.quaternary};
		> svg {
			width: 2rem;
			height: 2rem;
			color: ${theme.colors.secondary};
		}
	`}
`;
