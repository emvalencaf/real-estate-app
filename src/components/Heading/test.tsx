import { screen } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import Heading from ".";
import { renderTheme } from "../../styles/render-theme";
import { theme } from "../../styles/theme";

describe("<Heading />", () => {
	it("should render with default values", () => {
		renderTheme(<Heading>texto</Heading>);
		const heading = screen.getByRole("heading", { name: "texto" });

		expect(heading).toHaveStyle({
			color: theme.colors.primary,
			"font-size": theme.fonts.sizes.xhuge,
			"text-transform": "none",
		});
	});

	it("should render with white color", () => {
		renderTheme(<Heading>texto</Heading>);
		const heading = screen.getByRole("heading", { name: "texto" });

		expect(heading).toHaveStyle({
			color: theme.colors.white,
		});
	});

	it("should render correct heading sizes", () => {
		const { rerender } = renderTheme(
			<Heading size="small" as={"h6"} uppercase={false}>
				texto
			</Heading>
		);
		const heading = screen.getByRole("heading", { name: "texto" });

		expect(heading).toHaveStyle({
			"font-size": theme.fonts.sizes.medium,
		});

		rerender(
			<ThemeProvider theme={theme}>
				<Heading size="big" as={"h6"} uppercase={false}>
					texto
				</Heading>
			</ThemeProvider>
		);

		expect(screen.getByRole("heading", { name: "texto" })).toHaveStyle({
			"font-size": theme.fonts.sizes.xlarge,
		});

		rerender(
			<ThemeProvider theme={theme}>
				<Heading size="medium" as={"h6"} uppercase={false}>
					texto
				</Heading>
			</ThemeProvider>
		);

		expect(screen.getByRole("heading", { name: "texto" })).toHaveStyle({
			"font-size": theme.fonts.sizes.large,
		});

		rerender(
			<ThemeProvider theme={theme}>
				<Heading size="huge" as={"h6"} uppercase={false}>
					texto
				</Heading>
			</ThemeProvider>
		);

		expect(screen.getByRole("heading", { name: "texto" })).toHaveStyle({
			"font-size": theme.fonts.sizes.xhuge,
		});
	});

	it("should render correct font-size when using mobile", () => {
		renderTheme(
			<Heading size="huge" as={"h6"} uppercase={false}>
				texto
			</Heading>
		);
		screen.getByRole("heading", { name: "texto" });

		expect(screen.getByRole("heading", { name: "texto" })).toHaveStyleRule(
			"font-size",
			theme.fonts.sizes.xlarge,
			{
				media: theme.media.lteOrEqMedium,
			}
		);
	});

	it("should render with uppercase letters", () => {
		renderTheme(
			<Heading uppercase={true} as={"h6"} size={"big"}>
				texto
			</Heading>
		);
		const heading = screen.getByRole("heading", { name: "texto" });

		expect(heading).toHaveStyle({
			"text-transform": "uppercase",
		});
	});

	it("should render correct heading element", () => {
		const { container } = renderTheme(
			<Heading as="h6" size={"small"} uppercase={false}>
				texto
			</Heading>
		);
		screen.getByRole("heading", { name: "texto" });
		const h6 = container.querySelector("h6");

		expect(h6.tagName.toLowerCase()).toBe("h6");
	});

	it("should match snapshot", () => {
		const { container } = renderTheme(<Heading>texto</Heading>);

		expect(container).toMatchSnapshot();
	});
});
