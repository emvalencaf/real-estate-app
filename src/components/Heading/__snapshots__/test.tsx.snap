// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should match snapshot 1`] = `
.c0 {
  color: #0A1128;
  line-height: 1.2;
  font-size: 6.4rem;
  text-transform: none;
}

@media (max-width:360px) {
  .c0 {
    font-size: 3.2rem;
  }
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<div>
  <h1
    class="c0"
  >
    texto
  </h1>
</div>
`;
